pkg/certmanager/certmanager.go

```diff
@@ -84,29 +84,22 @@ func (c *CertManager) EnsureCertificate(vol *csiapi.MetaData, keyBundle *util.Ke
 		return nil, err
 	}
 
-	// Write metadata to file
-	metaDataBytes, err := json.Marshal(vol)
-	if err != nil {
-		return nil, err
-	}
-
-	metaPath := filepath.Join(vol.Path, csiapi.MetaDataFileName)
-	if err := ioutil.WriteFile(metaPath, metaDataBytes, 0600); err != nil {
+	if err := util.WriteMetaDataFile(vol); err != nil {
 		return nil, err
 	}
 
-	glog.V(4).Infof("cert-manager: metadata written to file %s", metaPath)
+	glog.V(4).Infof("cert-manager: metadata written to file %s", filepath.Join(vol.Path, csiapi.MetaDataFileName))
 
 	certPath := util.CertPath(vol)
 
-	if err := util.WriteFile(certPath, cr.Status.Certificate, 0600); err != nil {
+	if err := util.WriteFile(certPath, cr.Status.Certificate, (0644 | os.ModeSticky)); err != nil {
```

Comment:
Why do we set the sticky bit on the file? This man page suggests it will have no effect: https://man7.org/linux/man-pages/man1/chmod.1.html#RESTRICTED_DELETION_FLAG_OR_STICKY_BIT

Add an explanatory comment or remove it.

Reply:
Add an explanatory comment anyway, noting that the certificate needs to be world-readable.